handsontable/src/plugins/nestedHeaders/nestedHeaders.js

```diff
@@ -453,8 +461,22 @@ export class NestedHeaders extends BasePlugin {
       columnsToSelect.push(columnIndex, columnIndex + origColspan - 1, coords.row);
     }
 
-    // The plugin takes control of the how the columns are selected.
-    selection.selectColumns(...columnsToSelect);
+    // If there are no rendered rows, selecting columns causes a full headers redraw, blocking all the
+    // `click` events callbacks.
+    // In that case -> wait with the selecting action until all the other callbacks had been called by adding it at
+    // the top of the `click` event stack.
+    if (this.hot.countRenderedRows() === 0) {
+      const removeTempClickCallback = this.#eventManager.addEventListener(this.hot.rootElement, 'click', () => {
```

Comment:
The solution appears to be a temporary patch. I've investigated the issue, and it's a regression that I introduced within https://github.com/handsontable/handsontable/commit/08895d41f91294e5de8521b05423c951a0f8cf8d. After a couple of hours, I think I found the root cause of the problem and a more elegant solution - but it still has to be thoroughly tested.

Right now, when the table has no renderable cells at all, the table is always rendered in slow render mode. The headers are re-rendered each time the "mousedown" event occurs. At the same time, the "click" event that is registered by the _DropdownMenu_ plugin never had a chance to finish as the target element is destroyed. I specified the condition https://github.com/handsontable/handsontable/commit/8dd9b2b4783431c3c5b68e4c487a8adcd89959eb.

It would still be necessary to add tests to Walkontable - If you agree with the solution.

Reply:
I found that commit as well - I tried to reverse that behavior, so the table would render in the "fast" mode when there are no rows present, but as far as I remember, many test cases failed then, so I went with a more hacky approach. If you think that's the way to go, though, I will recheck it 👍